tests/core/policies/test_rule_policy.py

```diff
@@ -691,7 +691,7 @@ def assert_predicted_action(
 ) -> None:
     assert prediction.max_confidence == confidence
     index_of_predicted_action = prediction.max_confidence_index
-    prediction_action_name = domain.action_names[index_of_predicted_action]
+    prediction_action_name = domain.action_names_or_texts[index_of_predicted_action]
```

Comment:
can we please add tests for text predictions?

Reply:
especially the cases "loop wins over text prediction" should be covered and "text wins over default" should be covered.

Reply:
added both tests